projects/smoketest_max6643/src/main.c

```diff
@@ -0,0 +1,55 @@
+#include <stdbool.h>
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "max6643_fan_controller.h"
+#include "wait.h"
+// The smoke test (in a project called smoke_max6643) uses the MAX6643 driver
+// to register two separate callbacks which LOG_WARN a warning when the fan fail or
+// overtemp conditions are signalled.It also have a #define bool constant as a parameter
+// if that constant is true, it should turn the full
+// speed pin high, and if false, it should turn it low.
+
+#define FAN_FULL_SPEED true
+
+static void prv_fanfail_callback(const GpioAddress *address, void *context) {
+  LOG_WARN("Error %s\n", "Warning: The fan failed");
+}
+
+static void prv_overtemp_callback(const GpioAddress *address, void *context) {
+  LOG_WARN("Error %s\n", "Warning: Overtemp signal has been called");
+}
+
+int main(void) {
+  interrupt_init();
+  gpio_it_init();
+  gpio_init();
+  GpioAddress fanfail_pin_address = { .port = GPIO_PORT_B, .pin = 4 };
+  GpioAddress overtemp_pin_address = { .port = GPIO_PORT_B, .pin = 5 };
+  GpioAddress fan_pin_address = { .port = GPIO_PORT_B, .pin = 3 };
+  bool fan_speed_toggle = FAN_FULL_SPEED;
```

Comment:
Sorry, I meant you don't have to use these intermediary variables between the #defines and the Max6643Settings. You still need the defines, you just don't need to assign them to variables, so the settings struct would have lines like `.overtemp_pin  = OVERTEMP_PIN`.

Reply:
done